color/src/color.rs

```diff
@@ -630,20 +647,40 @@ impl<CS: ColorSpace> PremulColor<CS> {
         (d[0] * d[0] + d[1] * d[1] + d[2] * d[2] + d[3] * d[3]).sqrt()
     }
 
-    /// Pack into 8 bit per component encoding.
+    /// Convert the color to [sRGB][Srgb] if not already in sRGB, and pack into 8 bit per component
+    /// integer encoding.
+    ///
+    /// The RGBA components are mapped from the floating point range of `0.0-1.0` to the integer
+    /// range of `0-255`. Component values outside of this range are saturated to 0 or 255.
+    ///
+    /// # Implementation note
+    ///
+    /// This performs almost-correct rounding, see the note on [`AlphaColor::to_rgba8`].
     #[must_use]
     pub fn to_rgba8(self) -> PremulRgba8 {
-        // This does not need clamping as the behavior of a `f32` to `u8`
-        // cast in Rust is to saturate.
-        #[expect(clippy::cast_possible_truncation, reason = "deliberate quantization")]
         let [r, g, b, a] = self
             .convert::<Srgb>()
             .components
-            .map(|x| (x * 255.0).round() as u8);
+            .map(|x| fast_round_to_u8(x * 255.));
         PremulRgba8 { r, g, b, a }
     }
 }
 
+/// Fast rounding of `f32` to integer `u8`, rounding ties up.
+///
+/// Targeting x86, `f32::round` calls out to libc `roundf`. Even if that call were inlined, it is
+/// branchy, which would make it relatively slow. The following is faster, and on the range `0-255`
+/// almost correct*. AArch64 has dedicated rounding instructions so does not need this
+/// optimization, but the following is still fast.
+///
+/// * The only input where the output differs from `a.round() as u8` is `0.49999997`.
```

Comment:
Would it be reasonable to commit a test here? 

My suggested test would be a surgical once, which goes up in increments of a half and asserts that a few floats either side have the correct value (i.e. the same as `f32::round`, and if that test fails, `panic` unless the value is exactly `0.49999997`). That should be reasonably quick to run, and means that this claim doesn't need to be taken on trust.